ide/html.editor/src/org/netbeans/modules/html/editor/hints/other/AddMissingAltAttributeRule.java

```diff
@@ -0,0 +1,234 @@
+    }
+
+    private boolean isAltAttributeRequiredAndMissing(Document doc, OffsetRange range) {
+        OffsetRange adjusted = HtmlTagContextUtils.adjustContextRange(doc, range.getStart(), range.getEnd(), false);
+
+        return AddMissingAltAttributeRule.isAltAttributeRequired(doc, adjusted.getStart(), adjusted.getEnd()) && !AddMissingAltAttributeRule.isAltAttributeAvailable(doc, adjusted.getStart(), adjusted.getEnd());
+    }
+
+    private static boolean isAltAttributeRequired(final Document doc, final int from, final int to) {
+        final AtomicReference<List<Boolean>> result = new AtomicReference<>();
+
+        doc.render(() -> {
+            List<Boolean> found = new LinkedList<>();
+            result.set(found);
+            
+            TokenHierarchy th = TokenHierarchy.get(doc);
+            TokenSequence<HTMLTokenId> ts = Utils.getJoinedHtmlSequence(th, from);
+            
+            if (ts == null) {
+                //XXX - try to search backward and forward to find some html code???
+                return;
+            }
+            
+            //the joined ts is moved to the from offset and a token already selected (moveNext/Previous() == true)
+            //seek for all tag's attributes with css embedding representing an inlined style
+            String tag = null;
+            
+            do {
+                Token<HTMLTokenId> t = ts.token();
+                if (t.id() == HTMLTokenId.TAG_OPEN && t.text().toString().matches("img|applet|area")) {
+                    tag = t.text().toString();
+                } else if (t.id() == HTMLTokenId.TAG_CLOSE_SYMBOL) {
+                    //closing tag, produce the info
+                    if (tag != null) {
+                        // no alt tag found but required tag
+                        found.add(true);
+                        tag = null;
+                    }
+                }
+            } while (ts.moveNext() && ts.offset() <= to);
+        });
+
+        return !result.get().isEmpty();
+    }
+
+    private static boolean isAltAttributeAvailable(final Document doc, final int from, final int to) {
+        final AtomicReference<List<InlinedStyleInfo>> result = new AtomicReference<>();
+
+        doc.render(() -> {
+            List<InlinedStyleInfo> found = new LinkedList<>();
+            result.set(found);
+            
+            TokenHierarchy th = TokenHierarchy.get(doc);
+            TokenSequence<HTMLTokenId> ts = Utils.getJoinedHtmlSequence(th, from);
+            
+            if (ts == null) {
+                //XXX - try to search backward and forward to find some html code???
+                return;
+            }
+            
+            //the joined ts is moved to the from offset and a token already selected (moveNext/Previous() == true)
+            //seek for all tag's attributes with css embedding representing an inlined style
```

Comment:
This looks to complex to find the alt Attribute.  The variables here indicate extraction of class, alt, id - is this copied from somewhere else?

Reply:
Yes, also as you can see the comments, it was copied from the method findInlineStyle of class ExtractInlinedStyleRule. I started to copy stuff around, test some stuff and didn't fix code that much what was already there. Sry for that. I just wanted to get it working first. I will try to find anouther solution maybe with that what I already used for the HtmlCompletionProvider ` LexerUtils.followsToken(...`.